bob_the_builder/src/pkg_build.rs

```diff
@@ -0,0 +1,91 @@
+use std::{fs, path::Path, process::Command};
+
+use serde::Deserialize;
+
+use crate::cargo_toml::package::{Build, BuildSettings};
+
+#[derive(Deserialize, Debug, Eq, PartialEq)]
+pub struct ParsedPackage {
+    pub name: String,
+    pub builds: Vec<Build>,
+    pub default_build: bool,
+}
+
+impl ParsedPackage {
+    /// Build a contract with all the requested builds defined in `[package.metadata.optimizer]`
+    pub fn build(self, path: &Path) {
+        // Build all the requested builds
+        for build in self.builds.into_iter() {
+            build.build(path, &self.name);
+        }
+
+        if self.default_build {
+            // build contract without features or appended name
+            Build::default().build(path, &self.name);
+        }
+    }
+}
+
+impl Build {
+    /// Build the contract at the path *contract*.
+    pub fn build(self, contract: &Path, package_name: &str) {
+        let Build {
+            name: build_name,
+            settings: BuildSettings { features },
+        } = self;
+
+        let features = features.unwrap_or_default();
+
+        eprintln!("Building {} with features {:?}", package_name, features);
+
+        let mut args = vec![
+            "build",
+            "--release",
+            "--lib",
+            "--target-dir=/target",
+            "--target=wasm32-unknown-unknown",
+            "--locked",
+        ]
+        .into_iter()
+        .map(|arg| arg.to_string())
+        .collect::<Vec<String>>();
+
+        // Add features to command
+        let features_arg = features.into_iter().collect::<Vec<String>>().join(", ");
+        args.push(format!("--features={}", features_arg));
```

Comment:
Oh, @aumetra do you mean the ability to combine this with `--no-default-features`? I don't think it is possible yet but could be added.

The default build flag is something different

Reply:
Yeah, I just saw this comment above

> for the default-build, do you want the ability to build without default features or does the current solution work for you?

And didn't see a reply, so I wasn't quite sure where the consensus was, if it would be desirable to have an ability to toggle the default features off

Reply:
I think those are two different discussions. One is Cargo's `--no-default-features` (not implemented but would be good). The other is a `[package.metadata.optimizer]` default build (currently implemented).

Reply:
Then I misunderstood what was meant by the message. But yes, I agree that `--no-default-features` would be a good option to have